internal/database/db.go

```diff
@@ -0,0 +1,73 @@
+package database
+
+import (
+	"fmt"
+	"github.com/cyneruxyz/address-book/gen/proto"
+	m "github.com/cyneruxyz/address-book/internal/database/model"
+	"github.com/spf13/viper"
+	"gorm.io/gorm"
+)
+
+var (
+	conf *viper.Viper
+
+	model = &m.Fields{}
+	DSN   = fmt.Sprintf(
+		"user=%s password=%s dbname=%s port=%s sslmode=%s TimeZone=%s",
+		conf.GetString("DB_USER"),
+		conf.GetString("DB_PASSWORD"),
+		conf.GetString("DB_NAME"),
+		conf.GetString("DB_PORT"),
+		conf.GetString("DB_SSLMODE"),
+		conf.GetString("DB_TIMEZONE"),
+	)
+)
+
+type Database struct {
+	ORM *gorm.DB
+}
+
+func (db *Database) CreateItem(field *proto.AddressField) error {
+	return db.ORM.Create(model.Prepare(field)).Error
+}
+
+func (db *Database) ReadItem(param string) (fields []*proto.AddressField, err error) {
+	var items []m.Fields
+
+	err = db.ORM.Model(model).
+		Where("name LIKE ?", param).
+		Or("address LIKE ?", param).
+		Or("phone LIKE ?", param).
+		Limit(100).Find(items).Error
```

Comment:
Hardcoded value `100`. It should be configurable via flags, or at least be changed to var/const